lib/view_component/base.rb

```diff
@@ -648,7 +648,7 @@ def initialize_parameter_names
       end
 
       def initialize_parameters
-        instance_method(:initialize).parameters
+        @initialize_parameters ||= instance_method(:initialize).parameters
```

Comment:
Please set `@initialize_parameters = nil` in `.with_collection_parameter` above to cover the case where the collection parameter is changed after memoization.

Reply:
@camertron `initialize_parameters` is not inherited. I don't think there is a need to set it to `nil`. Even if `.with_collection_parameter` is called the `initialize` method was already defined in the first place. Can you plz give me an example to reproduce the issue you're talking about?

Reply:
Consider this (admittedly contrived) example:

```ruby
module Primer
  class ProductComponent < Primer::Component
    def initialize(product: nil)
      @product = product
    end
  end
end

# ensure parameters are memoized
Primer::ProductComponent.send(:initialize_parameters)

Primer::ProductComponent.class_eval do
  with_collection_parameter :item

  def initialize(item: nil)
    @item = item
  end
end
```

Trying to render this component as a collection will raise a `MissingCollectionArgumentError`.

Reply:
@camertron Got it. It's ready to merge now.